### ISSUE TITLE
Version "0.0.0-3.0.614" was chosen to replace version "2.4.606"

### ISSUE THREAD

**Issue by boris-faniuk-n26** [dependabot/dependabot-core#10311]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package ecosystem

Gradle

### Package manager version

8.7

### Manifest location and content before the Dependabot update

/build.gradle.kts

### dependabot.yml content

version: 2

registries:
  artifactory-gradle:
    type: maven-repository
    url: {some-url}
    username: ${{secrets.DEPENDABOT_ARTIFACTORY_USER}}
    password: ${{secrets.DEPENDABOT_MAVEN_ARTIFACTORY}}
    replaces-base: true

updates:
  - package-ecosystem: gradle
    directory: /
    groups:
      all-minor-and-patch:
        patterns:
          - "*"
        update-types:
          - "minor"
          - "patch"
    registries:
      - artifactory-gradle
    schedule:
      interval: daily

  - package-ecosystem: github-actions
    directory: /
    schedule:
      interval: daily
    groups:
      all-minor-and-patch:
        patterns:
          - "*"
        update-types:
          - "minor"
          - "patch"

### What you expected to see, versus what you actually saw

Current version of the dependency was "2.4.606".
New dependency release has got a version "0.0.0-3.0.614" with intent to avoid dependabot PRs.
Version "0.0.0-3.0.614" has major version "0" and should not replace version "2.4.606" that has major version "2".
However dependabot did create a PR to replace "2.4.606" with "0.0.0-3.0.614".

**Comment by abdulapopoola** [comment]
Thanks @boris-faniuk-n26 , this is odd. Tagging @amazimbe who is starting to work in this area as FYI.

Also tagging @jonjanego as FYI

**Comment by yeikel** [comment]
Can you please add a copy of `build.gradle.kts`? or at least some information with a reproducer?

**Comment by boris-faniuk-n26** [comment]
Hello, @yeikel ! 
Attaching the whole file is obviously not an option, I would have to cleanup a big part of it. 
Is there anything specific that may help reproducing?

**Comment by yeikel** [comment]
> Hello, @yeikel ! 
> Attaching the whole file is obviously not an option, I would have to cleanup a big part of it. 
> Is there anything specific that may help reproducing?

Yes, I think that the repository configuration and a snippet of your dependency definition is a good start( full coordinates would help)

Are you using any private registry or only Maven Central?

**Comment by jonjanego** [comment]
looks like dependabot is not aware of the version naming standards for maven dependencies, although it is admittedly a complicated standard.

here is the reference doc: https://maven.apache.org/pom.html#version-order-specification

**Comment by v-thavaahariharangit** [comment]
It's a feature request

<img width="1590" height="901" alt="Image" src="https://github.com/user-attachments/assets/02edbd59-815d-4d48-b578-78dfce31f01b" />

**Comment by yeikel** [comment]
I think that https://github.com/dependabot/dependabot-core/pull/13818 fixed this
